examples/function_todomvc/src/main.rs

```diff
@@ -0,0 +1,279 @@
+                    editing: false,
+                });
+                State {
+                    entries,
+                    filter: prev.filter,
+                }
+            }
+            Action::Remove(id) => {
+                let mut entries = prev.entries.clone();
+                entries.retain(|entry| entry.id != id);
+                State {
+                    entries,
+                    filter: prev.filter,
+                }
+            }
+            Action::Toggle(id) => {
+                let mut entries = prev.entries.clone();
+                let entry = entries.iter_mut().find(|entry| entry.id == id);
+                if let Some(entry) = entry {
+                    entry.completed = !entry.completed;
+                }
+                State {
+                    entries,
+                    filter: prev.filter,
+                }
+            }
+            Action::Edit((id, description)) => {
+                let mut entries = prev.entries.clone();
+
+                if description.is_empty() {
+                    entries.retain(|entry| entry.id != id)
+                }
+
+                let entry = entries.iter_mut().find(|entry| entry.id == id);
+                if let Some(entry) = entry {
+                    entry.description = description;
+                    entry.editing = false;
+                }
+                State {
+                    entries,
+                    filter: prev.filter,
+                }
+            }
+            Action::ToggleEdit(id) => {
+                let mut entries = prev.entries.clone();
+                let entry = entries.iter_mut().find(|entry| entry.id == id);
+                if let Some(entry) = entry {
+                    entry.editing = !entry.editing;
+                }
+                State {
+                    entries,
+                    filter: prev.filter,
+                }
+            }
+            Action::ToggleAll => {
+                let mut entries = prev.entries.clone();
+                for entry in &mut entries {
+                    if prev.filter.fits(entry) {
+                        entry.completed = !entry.completed;
+                    }
+                }
+                State {
+                    entries,
+                    filter: prev.filter,
+                }
+            }
+            Action::ClearCompleted => {
+                let mut entries = prev.entries.clone();
+                entries.retain(|e| Filter::Active.fits(e));
+                State {
+                    entries,
+                    filter: prev.filter,
+                }
+            }
+            Action::SetFilter(filter) => State {
+                filter,
+                entries: prev.entries.clone(),
+            },
+        },
+        // Initial state
+        State {
+            entries: LocalStorage::get(KEY).unwrap_or_else(|_| vec![]),
+            filter: Filter::All, // TODO: get from uri
+        },
```

Comment:
Seems as of right now the use_reducer type does not allow for anything other than Rc<State> see:
https://github.com/yewstack/yew/blob/78bb994ba23c9aa8d0bcfa8da2ddc972db5892b8/packages/yew/src/functional/hooks/use_reducer.rs#L71

Reply:
https://github.com/yewstack/yew/blob/78bb994ba23c9aa8d0bcfa8da2ddc972db5892b8/packages/yew/src/functional/hooks/use_reducer.rs#L65-L72

`Action`, `Reducer`, `State` are all generic types - it's unfortunate here that we also use a struct called `State` in the todo example but the `State` in `use_reducer` can be any type.

You could re-write the signature of `use_reducer` like this:
```rust
pub fn use_reducer<A, R, S>(
    reducer: R,
    initial_state: S,
) -> UseReducerHandle<S, A>
where
    A: 'static,
    R: Fn(Rc<S>, A) -> S + 'static,
    S: 'static,
```
and it would mean the same thing.